packages/angular_devkit/build_angular/src/angular-cli-files/plugins/karma.ts

```diff
@@ -72,8 +73,13 @@ const init: any = (config: any, emitter: any, customFileHandlers: any) => {
 
   // When using code-coverage, auto-add coverage-istanbul.
   config.reporters = config.reporters || [];
-  if (options.codeCoverage && config.reporters.indexOf('coverage-istanbul') === -1) {
-    config.reporters.push('coverage-istanbul');
+  // 'coverage-istanbul' comes from karma-coverage-istanbul-reporter package.
```

Comment:
We need to add the report based on the plugin which is configured in the karma.config.js. When users, have configured `karma-coverage-istanbul-reporter`, we need to add `coverage-istanbul`

Something like;
```ts
  // When using code-coverage, auto-add coverage-istanbul.
  if (options.codeCoverage) {
    // Determine which of the supported coverage reporter is being used.
    const pluginIds: string[] = [];
    config.plugins.forEach((p: object) => pluginIds.push(...Object.keys(p)));
   //  karma-coverage
    const hasKarmaCoveragePlugin = pluginIds.includes('reporter:coverage');
   //  karma-coverage-istanbul-reporter
    const hasKarmaIstanbulCoveragePlugin = pluginIds.includes('reporter:coverage-istanbul');

    // @keen - Maybe we shoudl display a warning, if we indent to deprecate this?
    if (hasKarmaIstanbulCoveragePlugin) {
      logger.warn(`Usage of 'karma-coverage-istanbul-reporter' is deprecated since version 10. Please use 'karma-coverage' instead.`)
    }

    if (hasKarmaCoveragePlugin && config.reporters.includes('coverage')) {
      config.reporters.push('coverage');
    } else if (hasKarmaIstanbulCoveragePlugin && config.reporters.includes('coverage-istanbul')) {
      config.reporters.push('coverage-istanbul');
    } else {
      // @keen - Maybe throw that we cannot determine which code coverage reporter is being used?
    }
  }

```